ui/flow_of_allowing_to_add_models.md

```diff
@@ -0,0 +1,123 @@
+This guide will tell you how to add a button to the list view that creates new instances of the item your application record represents.\
+In this guide we will implement the actual creation of the record in an EMS provider.\
+If creation of your items doesn't require interaction with an EMS API, you can adjust the process described here to implement it in manageiq core instead.
+
+The guide assumes that 
+* Your application-record is in `app/models/your_item.rb`
+* Your application-record belongs to an EMS. Usually its instances appear in MIQ by being refreshed from the EMS.
+* Your ui controller is in `plugins/manageiq-ui-classic/app/controllers/your_item_controller.rb`
+* Your provider is in `plugins/manageiq-providers-your-provider`
+* A menu for your item's list view is in `plugins/manageiq-ui-classic/app/heplers/application_helpers/toolbar/your_items_center.rb`
+
+You will with the following repositories:
+* manageiq
+* manageiq-ui-classic 
+* manageiq-api
+* manageiq-schema 
+* manageiq-providers-your-provider (if you implement the creation in an EMS)
+
+## Application Record
+### Base Application Record
+Make sure that the base application-record in `app/models/your_item.rb` already has `self.create_your_item`, `self.raw_create_your_item`, and `self.create_your_item_queue`.\
+If not, add these methods to the base record by copying them from some other model in `app/models/`.
+
+### Provider Application Record
+Make sure you have an application-record in `plugins/manageiq-providers-your-provider/app/models/manageiq/providers/your-provider/your_item.rb`.
+It should extend `app/models/your_item.rb`. 
+
+### Application Record type
+Make sure the database table for the application-record has a "type" column, so that STI can work properly.\
+If it doesn't have such a column, you'll need to add a migration file to `manageiq-schema` and create the column.    
+It will look something like
+```ruby
+class AddTypeToYourItems < ActiveRecord::Migration[6.0]
+  def change
+    add_column :your_items, :type, :string
+  end
+end
+```
+
+### Support create
+In the application-record in your provider, you need to add two entries.
+1. `supports :create`
+2. `def self.raw_create_your_item(ext_management_system, options = {})` 
+
+The method `self.raw_create_your_item` performs the actual work of creating.\  
+The options dict will contain what you get from your JS form.\
+In this method you need to call your EMS's API to create a real-world instance of your item.\
+You need to check if the request went through OK. If not, you need to throw an exception.
+If the request went OK, you will want to schedule a refresh so that the change will be reflected in MIQ DB.
+
+## Adding support for the new operation 
+1. In `app/models/ext_management_system.rb` add `supports_attribute :supports_create_your_item, :child_model => "YourItem"`
+2. In `manageiq-providers-autosde/app/models/manageiq/providers/your-provider/your_manager.rb`, add `supports :add_your_item`
```

Comment:
`manageiq-providers-autosde` => best to leave this generic as well like `manageiq-providers-your_provider`

Reply:
Actually with ``supports_attribute :supports_create_your_item, :child_model => "YourItem"` the correct way to do this would be:
`manageiq-providers-your_provider/app/models/manageiq/providers/your-provider/your_manager/your_thing.rb` and add `supports :create`